DateUtility.js

```diff
@@ -0,0 +1,239 @@
+      break;
+    case 4:
+      monthName = 'May';
+      break;
+    case 5:
+      monthName = 'June';
+      break;
+    case 6:
+      monthName = 'July';
+      break;
+    case 7:
+      monthName = 'August';
+      break;
+    case 8:
+      monthName = 'September';
+      break;
+    case 9:
+      monthName = 'October';
+      break;
+    case 10:
+      monthName = 'November';
+      break;
+    case 11:
+      monthName = 'December';
+      break;
+    default:
+      monthName = '';
+      break;
+    }
+	
+    return monthName;
+  }
+
+  /***************************************************************************
+  * Allows the internally stored date (i.e. "private") to be changed to
+  * what the user passes in. This method should be able to take either a date
+  * instance, a millisecond value, or nothing. If a date instance is passed
+  * in, set the interally stored date to a new instance based on this
+  * passed-in date. Do not store a reference to what is passed in.
+  * If nothing is passed in, set the internally stored date to "now".
+  ***************************************************************************/
+  function setDatePublic(aDate) {
+    if (arguments.length === 1) {
+      if (aDate instanceof Date) {
+        assignDate(aDate.getTime());
+      } else {
+        var tempDate = Number(aDate);
+		
+        if ((!Number.isNaN(tempDate)) && (tempDate > 0)) {
+          assignDate(tempDate);
+        } else {
+          assignDate(Date.now());
+        }
+      }
+    } else {
+      assignDate(Date.now());
+    }
+  }
+
+  /***************************************************************************
+  * Returns a boolean true or false, based on whether the internally stored
+  * date is in the "future" based on when this method is called.  This
+  * method takes no parameter.
+  ***************************************************************************/
+  function isFuturePublic() {
+    if ((datePrivate) && (datePrivate.getTime() > Date.now())) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /***************************************************************************
+  * Returns a boolean true or false, based on whether the internally stored
+  * date is "today" based on when this method is called. Note that this
+  * means the internally stored date can be in the future or the past
+  * relative to when this method is called.
+  ***************************************************************************/
+  function isTodayPublic() {
+    var today = new Date();
+
+    if ((datePrivate) &&
+        (datePrivate.getMonth() === today.getMonth()) &&
+        (datePrivate.getDate() === today.getDate()) &&
+        (datePrivate.getDay() === today.getDay()) &&
+        (datePrivate.getFullYear() === today.getFullYear())) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  // Return an object with the public methods
+  return {
+    getDate: getDatePublic
```

Comment:
This isn't quite the revealing module pattern in that your internal functions are named differently than the interface the module returns, but it still functions perfectly.